File: vipclaw-admin/src/main/java/com/vipamp/vipclaw/admin/dto/SkillResponse.java
package com.vipamp.vipclaw.admin.dto;

import com.vipamp.vipclaw.admin.entity.Skill;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 技能响应 DTO
 *
 * @author vipamp
 * @since 2026-03-16
 */
@Data
@Schema(description = "技能响应对象")
public class SkillResponse {

    /**
     * ID
     */
    @Schema(description = "ID", example = "1")
    private Long id;

    /**
     * 技能名称
     */
    @Schema(description = "技能名称", example = "test-skill")
    private String name;

    /**
     * 仓库ID
     */
    @Schema(description = "仓库ID", example = "1")
    private Long repositoryId;

    /**
     * 仓库名称
     */
    @Schema(description = "仓库名称", example = "qoder-skills")
    private String repositoryName;

    /**
     * 技能描述
     */
    @Schema(description = "技能描述")
    private String description;

    /**
     * skill.md 内容
     */
    @Schema(description = "skill.md 内容")
    private String skillmd;

    /**
     * 资源信息
     */
    @Schema(description = "资源信息")
    private String resources;

    /**
     * 是否启用（0:禁用，1:启用）
     */
    @Schema(description = "是否启用（0:禁用，1:启用）", example = "1")
    private Integer status;

    /**
     * 是否公开（0:否，1:是）
     */
    @Schema(description = "是否公开（0:否，1:是）", example = "1")
    private Integer isPublic;

    /**
     * 创建人
     */
    @Schema(description = "创建人", example = "admin")
    private String creator;

    /**
     * 创建时间
     */
    @Schema(description = "创建时间", example = "2026-03-16 12:00:00")
    private LocalDateTime createTime;

    /**
     * 更新时间
     */
    @Schema(description = "更新时间", example = "2026-03-16 12:00:00")
    private LocalDateTime updateTime;

    /**
     * 从实体对象转换
     *
     * @param skill 技能实体
     * @return 技能响应对象
     */
    public static SkillResponse fromEntity(Skill skill) {
        if (skill == null) {
            return null;
        }
        SkillResponse response = new SkillResponse();
        response.setId(skill.getId());
        response.setName(skill.getName());
        response.setRepositoryId(skill.getRepositoryId());
        response.setDescription(skill.getDescription());
        response.setSkillmd(skill.getSkillmd());
        response.setResources(skill.getResources());
        response.setStatus(skill.getStatus());
        response.setIsPublic(skill.getIsPublic());
        response.setCreator(skill.getCreator());
        response.setCreateTime(skill.getCreateTime());
        response.setUpdateTime(skill.getUpdateTime());
        return response;
    }
}

File: vipclaw-admin/src/main/kotlin/com/vipamp/vipclaw/admin/dto/SkillResponse.kt
package com.vipamp.vipclaw.admin.dto

import com.vipamp.vipclaw.admin.entity.Skill
import com.vipamp.vipclaw.admin.entity.SkillRepository
import io.swagger.v3.oas.annotations.media.Schema
import java.time.LocalDateTime

/**
 * Skill response object
 */
@Schema(description = "Skill response object")
data class SkillResponse(
    @Schema(description = "ID", example = "1")
    val id: Long? = null,
    @Schema(description = "Skill name", example = "test-skill")
    val name: String? = null,
    @Schema(description = "Repository ID", example = "1")
    val repositoryId: Long? = null,
    @Schema(description = "Repository name", example = "qoder-skills")
    var repositoryName: String? = null,
    @Schema(description = "Repository URL", example = "https://github.com/example/repo")
    var repositoryUrl: String? = null,
    @Schema(description = "Branch name", example = "main")
    var repositoryBranch: String? = null,
    @Schema(description = "Skill description")
    val description: String? = null,
    @Schema(description = "skill.md content")
    val skillmd: String? = null,
    @Schema(description = "Resource information")
    val resources: String? = null,
    @Schema(description = "Status (0:disabled, 1:enabled)", example = "1")
    val status: Int? = null,
    @Schema(description = "Whether public (0:no, 1:yes)", example = "1")
    val isPublic: Int? = null,
    @Schema(description = "Creator", example = "admin")
    val creator: String? = null,
    @Schema(description = "Creation time", example = "2026-03-16 12:00:00")
    val createTime: LocalDateTime? = null,
    @Schema(description = "Update time", example = "2026-03-16 12:00:00")
    val updateTime: LocalDateTime? = null,
) {
    companion object {
        /**
         * Convert Skill entity to SkillResponse
         */
        @JvmStatic
        fun fromEntity(
            skill: Skill?,
            repository: SkillRepository? = null,
        ): SkillResponse {
            if (skill == null) {
                return SkillResponse()
            }
            return SkillResponse(
                id = skill.id,
                name = skill.name,
                repositoryId = skill.repositoryId,
                repositoryName = repository?.name,
                repositoryUrl = repository?.url,
                repositoryBranch = repository?.branch,
                description = skill.description,
                skillmd = skill.skillmd,
                resources = skill.resources,
                status = skill.status,
                isPublic = skill.isPublic,
                creator = skill.creator,
                createTime = skill.createTime,
                updateTime = skill.updateTime,
            )
        }
    }
}
